pkg/dockerfile/generator.go

```diff
@@ -131,9 +133,62 @@ func (g *Generator) baseImage() (string, error) {
 }
 
 func (g *Generator) preamble() string {
-	return `ENV DEBIAN_FRONTEND=noninteractive
+	preamble := `ENV DEBIAN_FRONTEND=noninteractive
 ENV PYTHONUNBUFFERED=1
 ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu:/usr/local/nvidia/lib64:/usr/local/nvidia/bin`
+
+	envVarLines := g.envVariables()
+	if len(envVarLines) > 0 {
+		preamble += "\n" + strings.Join(envVarLines, "\n")
+	}
+
+	return preamble
+}
+
+func (g *Generator) envVariables() []string {
+	// If given, variables should be a list of strings, formatted like `KEY=VALUE`.
+	environmentVariables := g.Config.Build.BuildEnv
+
+	// Regex for valid environment variable names
+	regexpVariableName := regexp.MustCompile("^[A-Za-z_][A-Za-z0-9_]*$")
+
+	// Parse them into a list of strings. Also keep the ordered list of keys.
+	envVarKeys := []string{}
+	envVarMap := make(map[string]string)
+	if len(environmentVariables) > 0 {
+		envVarMap = make(map[string]string)
+		for _, v := range environmentVariables {
+			parts := strings.SplitN(v, "=", 2)
+
+			if ok := regexpVariableName.MatchString(parts[0]) && len(parts) == 2; ok {
+				envVarMap[parts[0]] = parts[1]
+				envVarKeys = append(envVarKeys, parts[0])
+			} else {
+				fmt.Printf("Ignoring invalid environment variable: %s\n", v)
```

Comment:
I wonder if this should be a hard error?

I wonder if we can just defer to Docker for this validation, and just put blindly put in the string into `ENV`?

Reply:
I'm OK with either of those options as well. I'll try them out and compare the experience

Reply:
Made it hard-error.